src/Tempest/Mapper/src/Mappers/ArrayToObjectMapper.php

```diff
@@ -160,9 +160,9 @@ private function resolveValueFromArray(
 
         $caster = $this->casterFactory->forProperty($property);
 
-        foreach ($data as $item) {
+        foreach ($data as $key => $item) {
             if (! is_array($item)) {
-                $values[] = $caster?->cast($item) ?? $item;
+                $values[$key] = $caster?->cast($item) ?? $item;
```

Comment:
Why was this change necessary?

Reply:
this is necessary when you have a input file with a name, so this ensures that the array is associative, please check the related issue.